### PR TITLE
Alert dev #3

### PR DESCRIPTION
**Pull request by lucas-m98**
Fixes to https://github.com/briweinstein/OpenWrt-IoT-IDS-Privacy/issues/17#issue-837232491

### CODE REVIEW

File: src/dashboard/luasrc/view/ids-priv/config.htm [hunk 1]
```diff
@@ -131,14 +131,15 @@
...
+					config_count = config_count + 1
 				end
 			end
 		end
 		return_string = return_string .. "</table>"
 		return return_string
 	end
 
-	local data = format_ip_neighbors(luci.sys.exec("ip neigh | grep -h \"192.\""))
+	local data = format_ip_neighbors(luci.sys.exec("ip neigh"))
```
Comment (lucas-m98) [inline src/dashboard/luasrc/view/ids-priv/config.htm:142]: Removed filter in case you want the ability to add more devices @MartinLPetrauskas

Comment (MartinLPetrauskas) [reply src/dashboard/luasrc/view/ids-priv/config.htm:142]: Thank you kind sir.

File: src/dashboard/alerts/alerts.json [hunk 2]
```diff
@@ -0,0 +1,125 @@
+{
```
Comment (MartinLPetrauskas) [inline src/dashboard/alerts/alerts.json:1]: Is this file needed? It seems like it is just an example file. We don't want to install unnecessary files on the router.

**Review by MartinLPetrauskas — CHANGES_REQUESTED** [review]
This should solve some of the problems on the config page. Only request is to remove the alerts.json file since we do not want to install unnecessary files.

File: src/dashboard/alerts/alerts.json (hunk 2 above)
Comment (lucas-m98) [reply src/dashboard/alerts/alerts.json:1]: Not necessary, thought it would be good to have examples in the repo but I can remove it.

**Review by MartinLPetrauskas — APPROVED** [review]
PR looks good. I don't forsee any issues when compiling this. since the changes aren't major

File: src/dashboard/alerts/alert.py [hunk 3]
```diff
@@ -145,12 +122,12 @@ def save_alert(self):
 
     def alert(self):
         """
-        Inform the townspeople (Send the alert where it should go)
```
Comment (briweinstein) [inline src/dashboard/alerts/alert.py:148]: Sad to see it go

**Review by briweinstein — APPROVED** [review]

File: src/dashboard/alerts/alert.py [hunk 4]
```diff
@@ -4,7 +4,7 @@
 import hashlib
```
Comment (samir-elhelw76) [inline src/dashboard/alerts/alert.py:4]: could not comment on the above two files under the "packet_analysis" directory but why are they here? Can they be removed since they are both blank?

File: src/dashboard/alerts/alert.py [hunk 5]
```diff
@@ -64,20 +40,24 @@ def __init__(self, pkt: Packet, alert_description="", alert_type=ALERT_TYPE.UNKN
...
+                    self.device_mac = pkt["Ethernet"].dst
             else:
-                self.device_mac = "[Unknown]"
+                if "IP" in pkt:
+                    self.device_ip = pkt["IP"].src
+                else:
+                    self.device_ip = "[Layer 2]"
+                if "Ethernet" in pkt:
+                    self.device_mac = pkt["Ethernet"].src
+                else:
```
Comment (samir-elhelw76) [inline src/dashboard/alerts/alert.py:56]: Can you explain why these checks are being done twice in this block of code? They can probably be refactored to simplify the code and make it more readable as well.

File: src/dashboard/alerts/alert.py [hunk 6]
```diff
@@ -64,20 +40,24 @@ def __init__(self, pkt: Packet, alert_description="", alert_type=ALERT_TYPE.UNKN
...
+                    self.device_mac = "[Unknown]"
+            self.payload_info = raw(pkt)
+        else:
+            self.payload_info = "None"
 
         # Use some magic config trickery to get the name for this device, otherwise unknown
         if "magic":
```
Comment (samir-elhelw76) [inline src/dashboard/alerts/alert.py:63]: What is this line supposed to be doing?

File: src/dashboard/luasrc/view/ids-priv/config.htm (hunk 1 above)
Comment (samir-elhelw76) [reply src/dashboard/luasrc/view/ids-priv/config.htm:142]: What is this system call doing?

File: src/dashboard/luasrc/controller/ids-priv/ids_tab.lua [hunk 7]
```diff
@@ -22,11 +21,11 @@ module("luci.controller.ids-priv.ids_tab", package.seeall)  --notice that new_ta
         local payload = luci.jsonc.parse(tostring(config))
 
         -- Open configuration file as JSON
-        local conf = io.open("/etc/capstone-ids/config.json", 'r')
+        local conf = io.open("/etc/tinyHIPPO/config.json", 'r')
```
Comment (samir-elhelw76) [inline src/dashboard/luasrc/controller/ids-priv/ids_tab.lua:24]: In the python portion of our package we created a config object to avoid having to read from the file everytime a config value needs to be accessed. Is there a similar mechanism for Luci? If so can it be implemented here and in other parts of your luci code?

**Review by samir-elhelw76 — CHANGES_REQUESTED** [review]

File: src/dashboard/luasrc/controller/ids-priv/ids_tab.lua (hunk 7 above)
Comment (lucas-m98) [reply src/dashboard/luasrc/controller/ids-priv/ids_tab.lua:24]: Well the config file is being directly altered in order to write the new MAC's to the file, so it kinda requires an I/O operation. This is also the sole place where the config file is being accessed so creating another object just to track use in one place seems redundant.

File: src/dashboard/alerts/alert.py (hunk 4 above)
Comment (lucas-m98) [reply src/dashboard/alerts/alert.py:4]: Those were from local testing, I can remove. I didn't see them added.

File: src/dashboard/alerts/alert.py (hunk 6 above)
Comment (lucas-m98) [reply src/dashboard/alerts/alert.py:63]: We wanted to have some form of a device name/classification (Camera, Light Strip 2, etc.) added to alerts for ease of use, and this is a placeholder for that. If we no longer plan on including this I can remove it, but I assumed we were still going down this route once we found some way to pull hostnames or resolve MAC addresses some how.

File: src/dashboard/luasrc/view/ids-priv/config.htm (hunk 1 above)
Comment (lucas-m98) [reply src/dashboard/luasrc/view/ids-priv/config.htm:142]: https://man7.org/linux/man-pages/man8/ip-neighbour.8.html

Comment (lucas-m98) [reply src/dashboard/luasrc/view/ids-priv/config.htm:142]: The `arp -a` command most commonly used in this instance isn't actually a callable shell command, so I had to use this as an alternative.

File: src/dashboard/alerts/alert.py (hunk 5 above)
Comment (lucas-m98) [reply src/dashboard/alerts/alert.py:56]: I can organize it to remove the inner else statements and just assume that every packet will have an _Ethernet_ segment (Hopefully that won't cause a crash, but honestly if we do get a packet without an Ethernet segment I'll be kinda amazed), but its still 2 layers deep of a binary decision tree so there are 4 outcomes and I can't really reduce the amount of checks for IP because they have independent outcomes.

Comment (briweinstein) [reply src/dashboard/alerts/alert.py]: Maybe just add a try/catch for edge cases

File: src/dashboard/alerts/alert.py (hunk 6 above)
Comment (samir-elhelw76) [reply src/dashboard/alerts/alert.py:59]: Can this just be replaced with a TODO: then so it flags in our IDEs instead of the conditional here?

File: src/dashboard/luasrc/controller/ids-priv/ids_tab.lua (hunk 7 above)
Comment (samir-elhelw76) [reply src/dashboard/luasrc/controller/ids-priv/ids_tab.lua:24]: If that is the case, once we have SQLite in-place this will be removed right? Additionally can the I/O operation done on the file below (line 42) be removed and can this declaration be modified to open the file for reading and writing?

**Review by samir-elhelw76 — CHANGES_REQUESTED** [review]
Had a couple of more comments to review

File: src/dashboard/luasrc/controller/ids-priv/ids_tab.lua (hunk 7 above)
Comment (lucas-m98) [reply src/dashboard/luasrc/controller/ids-priv/ids_tab.lua:24]: Sadly, Lua doesn't work like that. The closest file operation to an "update" (Read and write) is a combo of read and append ("r+"). So in order to rewrite the MAC values inline (Without appending the whole thing again), I need to read and write in two operations. But yes, eventually this should be migrated once we move over the python config file.

File: src/dashboard/alerts/alert.py (hunk 5 above)
Comment (lucas-m98) [reply src/dashboard/alerts/alert.py]: Can do. The only edge cases it could get would be malformed test packets from within our own tests, but I can add it

File: src/dashboard/alerts/alert.py (hunk 6 above)
Comment (lucas-m98) [reply src/dashboard/alerts/alert.py:59]: I can do that.

**Review by samir-elhelw76 — APPROVED** [review]
